server/src/controllers/song.controllers.ts

```diff
@@ -19,88 +19,81 @@ const createSong: RequestHandler = async (req: Request, res: Response): Promise<
         sendResponse(res, 404, 'Song not created');
       }
     } catch (error: any) {
-      sendResponse(res, 500, error.message);
+      sendResponse(res, 500, error?.message);
     }
   } else {
     sendResponse(res, 400, 'Missing required fields' );
   }
-  return '';
 };
 
-const getSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument[] | SongDocument | string> => {
+const getSong: RequestHandler = async (req: Request, res: Response): Promise<void> => {
     const { id: songId } = req.params;
 
     if (songId) {
       try {
-        const data = await Song.findOne({ _id: songId }).exec();
+        const data: SongDocument = await Song.findOne({ _id: songId, isDeleted: false }).exec();
   
         if (data) {
           sendResponse(res, 200, data);
         } else {
           sendResponse(res, 404, 'Song not found');
         }
       } catch (error: any) {
-        sendResponse(res, 500, error.message );
+        sendResponse(res, 500, error?.message );
       }
     } else {
       try {
-        const data = await Song.find().exec();
+        const data = await Song.find({ isDeleted: false }).exec();
   
         if (data) {
           sendResponse(res, 200, data);
         } else {
           sendResponse(res, 404, 'Songs not found');
         }
       } catch (error: any) {
-        sendResponse(res, 500, error.message);
+        sendResponse(res, 500, error?.message);
       }
     }
-
-    return '';
 };
 
-const updateSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument | string> => {
+const updateSong: RequestHandler = async (req: Request, res: Response): Promise<void> => {
     const { id: songId, ...toUpdate } = req.body;
 
     if (songId && Object.keys(toUpdate).length > 0) {
       try {
-        const updatedGroup = await Song.updateOne({ _id: songId }, { $set: toUpdate });
+        const updatedSong = await Song.updateOne({ _id: songId, isDeleted: false }, { $set: toUpdate });
   
-        if (updatedGroup) {
-          sendResponse(res, 200, 'Group updated');
+        if (updatedSong) {
+          sendResponse(res, 200, 'Song updated');
         } else {
-          sendResponse(res, 404, 'Group not updated');
+          sendResponse(res, 404, 'Song not updated');
         }
       } catch (error: any) {
-        sendResponse(res, 500, error.message);
+        sendResponse(res, 500, error?.message);
       }
     } else {
       sendResponse(res, 400, 'Missing required fields');
     }
-
-    return '';
 };
 
-const deleteSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument | string> => {
+const deleteSong: RequestHandler = async (req: Request, res: Response): Promise<void> => {
   const { id: songId } = req.params;
 
   if (songId) {
     try {
       const data = await Song.findOneAndDelete({ _id: songId }, { $set: { isDeleted: true } });
```

Comment:
same with yy's pr. Can define the type or better approach change it to updateOne.